### ISSUE TITLE
ISSUE INSTALLING SAMI IN LARAVEL5.2.3

### ISSUE THREAD

**Issue by goldenboi039** [FriendsOfPHP/Sami#241]
I am trying to write documentation for a laravel 5.2.3 project and I intend to use Sami to acheive this.

I followed the the git hub link https://github.com/FriendsOfPHP/Sami and I tried to install using composer

composer require sami/sami

But I keep getting the error,

Using version ^3.3 for sami/sami ./composer.json has been updated Loading composer repositories with package information Updating dependencies (including require-dev) Your requirements could not be resolved to an installable set of packages.

Problem 1 - Installation request for sami/sami ^3.3 -> satisfiable by sami/sami[v3.3.0]. - Conclusion: remove symfony/console v3.1.5 - Conclusion: don't install symfony/console v3.1.5 - sami/sami v3.3.0 requires symfony/console ~2.1 -> satisfiable by symfony/console[v2.1.0, v2.1.1, v2.1.10, v2.1.11, v2.1.12, v2.1.13, v2.1.2, v2.1.3, v2.1.4, v2.1.5, v2.1.6, v2.1.7, v2.1.8, v2.1.9, v2.2.0, v2.2.1, v2.2.10, v2.2.11, v2.2.2, v2.2.3, v2.2.4, v2.2.5, v2.2.6, v2.2.7, v2.2.8, v2.2.9, v2.3.0, v2.3.1, v2.3.10, v2.3.11, v2.3.12, v2.3.13, v2.3.14, v2.3.15, v2.3.16, v2.3.17, v2.3.18, v2.3.19, v2.3.2, v2.3.20, v2.3.21, v2.3.22, v2.3.23, v2.3.24, v2.3.25, v2.3.26, v2.3.27, v2.3.28, v2.3.29, v2.3.3, v2.3.30, v2.3.31, v2.3.32, v2.3.33, v2.3.34, v2.3.35, v2.3.36, v2.3.37, v2.3.38, v2.3.39, v2.3.4, v2.3.40, v2.3.41, v2.3.42, v2.3.5, v2.3.6, v2.3.7, v2.3.8, v2.3.9, v2.4.0, v2.4.1, v2.4.10, v2.4.2, v2.4.3, v2.4.4, v2.4.5, v2.4.6, v2.4.7, v2.4.8, v2.4.9, v2.5.0, v2.5.1, v2.5.10, v2.5.11, v2.5.12, v2.5.2, v2.5.3, v2.5.4, v2.5.5, v2.5.6, v2.5.7, v2.5.8, v2.5.9, v2.6.0, v2.6.1, v2.6.10, v2.6.11, v2.6.12, v2.6.13, v2.6.2, v2.6.3, v2.6.4, v2.6.5, v2.6.6, v2.6.7, v2.6.8, v2.6.9, v2.7.0, v2.7.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.3, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.2, v2.8.3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9]. - Can only install one of: symfony/console[v2.3.10, v3.1.5]. - Can only install one of: symfony/console[v2.3.11, v3.1.5]. - Can only install one of: symfony/console[v2.3.12, v3.1.5]. - Can only install one of: symfony/console[v2.3.13, v3.1.5]. - Can only install one of: symfony/console[v2.3.14, v3.1.5]. - Can only install one of: symfony/console[v2.3.15, v3.1.5]. - Can only install one of: symfony/console[v2.3.16, v3.1.5]. - Can only install one of: symfony/console[v2.3.17, v3.1.5]. - Can only install one of: symfony/console[v2.3.18, v3.1.5]. - Can only install one of: symfony/console[v2.3.19, v3.1.5]. - Can only install one of: symfony/console[v2.3.20, v3.1.5]. - Can only install one of: symfony/console[v2.3.21, v3.1.5]. - Can only install one of: symfony/console[v2.3.22, v3.1.5]. - Can only install one of: symfony/console[v2.3.23, v3.1.5]. - Can only install one of: symfony/console[v2.3.24, v3.1.5]. - Can only install one of: symfony/console[v2.3.25, v3.1.5]. - Can only install one of: symfony/console[v2.3.26, v3.1.5]. - Can only install one of: symfony/console[v2.3.27, v3.1.5]. - Can only install one of: symfony/console[v2.3.28, v3.1.5]. - Can only install one of: symfony/console[v2.3.29, v3.1.5]. - Can only install one of: symfony/console[v2.3.30, v3.1.5]. - Can only install one of: symfony/console[v2.3.31, v3.1.5]. - Can only install one of: symfony/console[v2.3.32, v3.1.5]. - Can only install one of: symfony/console[v2.3.33, v3.1.5]. - Can only install one of: symfony/console[v2.3.34, v3.1.5]. - Can only install one of: symfony/console[v2.3.35, v3.1.5]. - Can only install one of: symfony/console[v2.3.36, v3.1.5]. - Can only install one of: symfony/console[v2.3.37, v3.1.5]. - Can only install one of: symfony/console[v2.3.38, v3.1.5]. - Can only install one of: symfony/console[v2.3.39, v3.1.5]. - Can only install one of: symfony/console[v2.3.40, v3.1.5]. - Can only install one of: symfony/console[v2.3.41, v3.1.5]. - Can only install one of: symfony/console[v2.3.42, v3.1.5]. - Can only install one of: symfony/console[v2.4.10, v3.1.5]. - Can only install one of: symfony/console[v2.4.2, v3.1.5]. - Can only install one of: symfony/console[v2.4.3, v3.1.5]. - Can only install one of: symfony/console[v2.4.4, v3.1.5]. - Can only install one of: symfony/console[v2.4.5, v3.1.5]. - Can only install one of: symfony/console[v2.4.6, v3.1.5]. - Can only install one of: symfony/console[v2.4.7, v3.1.5]. - Can only install one of: symfony/console[v2.4.8, v3.1.5]. - Can only install one of: symfony/console[v2.4.9, v3.1.5]. - Can only install one of: symfony/console[v2.5.0, v3.1.5]. - Can only install one of: symfony/console[v2.5.1, v3.1.5]. - Can only install one of: symfony/console[v2.5.10, v3.1.5]. - Can only install one of: symfony/console[v2.5.11, v3.1.5]. - Can only install one of: symfony/console[v2.5.12, v3.1.5]. - Can only install one of: symfony/console[v2.5.2, v3.1.5]. - Can only install one of: symfony/console[v2.5.3, v3.1.5]. - Can only install one of: symfony/console[v2.5.4, v3.1.5]. - Can only install one of: symfony/console[v2.5.5, v3.1.5]. - Can only install one of: symfony/console[v2.5.6, v3.1.5]. - Can only install one of: symfony/console[v2.5.7, v3.1.5]. - Can only install one of: symfony/console[v2.5.8, v3.1.5]. - Can only install one of: symfony/console[v2.5.9, v3.1.5]. - Can only install one of: symfony/console[v2.6.0, v3.1.5]. - Can only install one of: symfony/console[v2.6.1, v3.1.5]. - Can only install one of: symfony/console[v2.6.10, v3.1.5]. - Can only install one of: symfony/console[v2.6.11, v3.1.5]. - Can only install one of: symfony/console[v2.6.12, v3.1.5]. - Can only install one of: symfony/console[v2.6.13, v3.1.5]. - Can only install one of: symfony/console[v2.6.2, v3.1.5]. - Can only install one of: symfony/console[v2.6.3, v3.1.5]. - Can only install one of: symfony/console[v2.6.4, v3.1.5]. - Can only install one of: symfony/console[v2.6.5, v3.1.5]. - Can only install one of: symfony/console[v2.6.6, v3.1.5]. - Can only install one of: symfony/console[v2.6.7, v3.1.5]. - Can only install one of: symfony/console[v2.6.8, v3.1.5]. - Can only install one of: symfony/console[v2.6.9, v3.1.5]. - Can only install one of: symfony/console[v2.7.0, v3.1.5]. - Can only install one of: symfony/console[v2.7.1, v3.1.5]. - Can only install one of: symfony/console[v2.7.10, v3.1.5]. - Can only install one of: symfony/console[v2.7.11, v3.1.5]. - Can only install one of: symfony/console[v2.7.12, v3.1.5]. - Can only install one of: symfony/console[v2.7.13, v3.1.5]. - Can only install one of: symfony/console[v2.7.14, v3.1.5]. - Can only install one of: symfony/console[v2.7.15, v3.1.5]. - Can only install one of: symfony/console[v2.7.16, v3.1.5]. - Can only install one of: symfony/console[v2.7.17, v3.1.5]. - Can only install one of: symfony/console[v2.7.18, v3.1.5]. - Can only install one of: symfony/console[v2.7.19, v3.1.5]. - Can only install one of: symfony/console[v2.7.2, v3.1.5]. - Can only install one of: symfony/console[v2.7.3, v3.1.5]. - Can only install one of: symfony/console[v2.7.4, v3.1.5]. - Can only install one of: symfony/console[v2.7.5, v3.1.5]. - Can only install one of: symfony/console[v2.7.6, v3.1.5]. - Can only install one of: symfony/console[v2.7.7, v3.1.5]. - Can only install one of: symfony/console[v2.7.8, v3.1.5]. - Can only install one of: symfony/console[v2.7.9, v3.1.5]. - Can only install one of: symfony/console[v2.8.0, v3.1.5]. - Can only install one of: symfony/console[v2.8.1, v3.1.5]. - Can only install one of: symfony/console[v2.8.10, v3.1.5]. - Can only install one of: symfony/console[v2.8.11, v3.1.5]. - Can only install one of: symfony/console[v2.8.12, v3.1.5]. - Can only install one of: symfony/console[v2.8.2, v3.1.5]. - Can only install one of: symfony/console[v2.8.3, v3.1.5]. - Can only install one of: symfony/console[v2.8.4, v3.1.5]. - Can only install one of: symfony/console[v2.8.5, v3.1.5]. - Can only install one of: symfony/console[v2.8.6, v3.1.5]. - Can only install one of: symfony/console[v2.8.7, v3.1.5]. - Can only install one of: symfony/console[v2.8.8, v3.1.5]. - Can only install one of: symfony/console[v2.8.9, v3.1.5]. - Can only install one of: symfony/console[v2.1.0, v3.1.5]. - Can only install one of: symfony/console[v2.1.1, v3.1.5]. - Can only install one of: symfony/console[v2.1.10, v3.1.5]. - Can only install one of: symfony/console[v2.1.11, v3.1.5]. - Can only install one of: symfony/console[v2.1.12, v3.1.5]. - Can only install one of: symfony/console[v2.1.13, v3.1.5]. - Can only install one of: symfony/console[v2.1.2, v3.1.5]. - Can only install one of: symfony/console[v2.1.3, v3.1.5]. - Can only install one of: symfony/console[v2.1.4, v3.1.5]. - Can only install one of: symfony/console[v2.1.5, v3.1.5]. - Can only install one of: symfony/console[v2.1.6, v3.1.5]. - Can only install one of: symfony/console[v2.1.7, v3.1.5]. - Can only install one of: symfony/console[v2.1.8, v3.1.5]. - Can only install one of: symfony/console[v2.1.9, v3.1.5]. - Can only install one of: symfony/console[v2.2.0, v3.1.5]. - Can only install one of: symfony/console[v2.2.1, v3.1.5]. - Can only install one of: symfony/console[v2.2.10, v3.1.5]. - Can only install one of: symfony/console[v2.2.11, v3.1.5]. - Can only install one of: symfony/console[v2.2.2, v3.1.5]. - Can only install one of: symfony/console[v2.2.3, v3.1.5]. - Can only install one of: symfony/console[v2.2.4, v3.1.5]. - Can only install one of: symfony/console[v2.2.5, v3.1.5]. - Can only install one of: symfony/console[v2.2.6, v3.1.5]. - Can only install one of: symfony/console[v2.2.7, v3.1.5]. - Can only install one of: symfony/console[v2.2.8, v3.1.5]. - Can only install one of: symfony/console[v2.2.9, v3.1.5]. - Can only install one of: symfony/console[v2.3.0, v3.1.5]. - Can only install one of: symfony/console[v2.3.1, v3.1.5]. - Can only install one of: symfony/console[v2.3.2, v3.1.5]. - Can only install one of: symfony/console[v2.3.3, v3.1.5]. - Can only install one of: symfony/console[v2.3.4, v3.1.5]. - Can only install one of: symfony/console[v2.3.5, v3.1.5]. - Can only install one of: symfony/console[v2.3.6, v3.1.5]. - Can only install one of: symfony/console[v2.3.7, v3.1.5]. - Can only install one of: symfony/console[v2.3.8, v3.1.5]. - Can only install one of: symfony/console[v2.3.9, v3.1.5]. - Can only install one of: symfony/console[v2.4.0, v3.1.5]. - Can only install one of: symfony/console[v2.4.1, v3.1.5]. - Installation request for symfony/console (locked at v3.1.5) -> satisfiable by symfony/console[v3.1.5].

Is Sami not compatible with Laravel 5.2.3? Because I also tried adding a lower sympony version(~2.6) but an error was displayed indicating the minimum sympony version for laravel 5.2.3 is 3.1.5.

What should i do?

**Comment by matthewfritz** [comment]
I was having this issue earlier today. What worked for me was this:

1. Deleting the vendor directory
2. Deleting the composer.lock file
3. Adding "sami/sami": "3.3.*" to the "require-dev" key in composer.json
4. Running composer install to install all the dependencies from scratch

Then I realized I didn't want to have to re-install Sami as a dependency in each project so I just downloaded the .phar file through cURL, placed it in my PATH, and then I was able to execute "sami" from anywhere on my filesystem.
